pyfr/plugins/turbulence.py

```diff
@@ -0,0 +1,270 @@
+# -*- coding: utf-8 -*-
+
+import math
+import numpy as np
+import random
+import time
+import uuid
+from rtree import index
+
+from collections import defaultdict
+from numpy.lib.recfunctions import structured_to_unstructured
+from pyfr.plugins.base import BaseSolverPlugin
+from pyfr.regions import BoxRegion
+from pyfr.mpiutil import get_comm_rank_root
+
+class pcg32rxs_m_xs:
+    def __init__(self, seed):
+        self.state = np.uint32(seed)
+        self.multiplier = np.uint32(747796405)
+        self.mcgmultiplier = np.uint32(277803737)
+        self.increment = np.uint32(2891336453)
+        self.opbits = np.uint8(4)
+        self.b22 = np.uint8(22)
+        self.b32 = np.uint8(32)
+    def rand(self):
+        oldstate = self.state
+        self.state = (oldstate * self.multiplier) + self.increment
+        rshift = np.uint8(oldstate >> (self.b32 - self.opbits))
+        oldstate ^= oldstate >> (self.opbits + rshift)
+        oldstate *= self.mcgmultiplier
+        oldstate ^= oldstate >> self.b22
+        return oldstate
+    def random(self):
+        # change to match kernel code exactly ?
+        return np.ldexp(self.rand(),-32)
+    def getstate(self):
+        return self.state
+    def randint(self, a, b):
+        return a + self.rand() % (b - a)
+    def choice(self, seq):
+        return seq[self.randint(0, len(seq))]
+        
+class Turbulence(BaseSolverPlugin):
+    name = 'turbulence'
+    systems = ['navier-stokes']
+    formulations = ['std']
+
+    def __init__(self, intg, cfgsect, suffix):
```

Comment:
It seems that his plugin only works for 3D problems, it might be worth adding a check for that.